src/main/java/com/bobocode/petros/container/DependencyDefinition.java

```diff
@@ -26,54 +29,74 @@ public DependencyDefinition() {
         injectedDependencyDefinitions = new ArrayList<>();
     }
 
-    public Object getDependencyClass() {
-        if (dependency == null) {
-            try {
-                dependency = Class.forName(qualifiedName).getConstructor().newInstance();
-            } catch (Exception e) {
-                LOG.error("No default constructor found for class {}", qualifiedName);
-                throw new NoDefaultConstructorException(qualifiedName);
-            }
-        }
-        return dependency;
-    }
-
+    /**
+     * Return true in case if Dependency is declared inside configuration class that is marked with ConfigClass annotation
+     */
     public boolean isConfigClassDependency() {
         return configClassDependency;
     }
 
+    /**
+     * Set the configClassDependency flag to true in case if Dependency is declared inside configuration class that is marked with ConfigClass annotation
+     */
     public void setConfigClassDependency(boolean configClassDependency) {
```

Comment:
Please use annotation @param to describe method parameters

Reply:
Done